ext/spl/php_spl.stub.php

```diff
@@ -33,3 +33,294 @@ function iterator_apply(Traversable $iterator, callable $function, ?array $args
 function iterator_count(Traversable $iterator): int {}
 
 function iterator_to_array(Traversable $iterator, bool $use_keys = true): array {}
+
+/* ext/spl/spl_array.c */
+
+class ArrayObject implements IteratorAggregate, ArrayAccess, Serializable, Countable
+{
+    /** @param array|object $input */
+    public function __construct($input = UNKNOWN, int $flags = 0, string $iterator_class = "ArrayIterator") {}
+
+    /**
+     * @param mixed $index
+     * @return bool
+     */
+    public function offsetExists($index) {}
+
+    /**
+     * @param mixed $index
+     * @return mixed
+     */
+    public function offsetGet($index) {}
+
+    /**
+     * @param mixed $index
+     * @param mixed $value
+     * @return void
+     */
+    public function offsetSet($index, $value) {}
+
+    /**
+     * @param mixed $index
+     * @return void
+     */
+    public function offsetUnset($index) {}
+
+    /**
+     * @param mixed $value
+     * @return void
+     */
+    public function append($value) {}
+
+    /** @return array */
+    public function getArrayCopy() {}
+
+    /** @return int */
+    public function count() {}
+
+    /** @return int */
+    public function getFlags() {}
+
+    /** @return void */
+    public function setFlags(int $flags) {}
+
+    /**
+     * @param int $sort_flags
+     * @return bool
+     */
+    public function asort($sort_flags = SORT_REGULAR) {}
+
+    /**
+     * @param int $sort_flags
+     * @return bool
+     */
+    public function ksort($sort_flags = SORT_REGULAR) {}
+
+    /**
+     * @param callback $cmp_function
+     * @return bool
+     */
+    public function uasort($cmp_function) {}
+
+    /**
+     * @param callback $cmp_function
+     * @return bool
+     */
+    public function uksort($cmp_function) {}
+
+    /** @return bool */
+    public function natsort() {}
+
+    /** @return bool */
+    public function natcasesort() {}
+
+    /** @return void */
+    public function unserialize(string $serialized) {}
+
+    /** @return string|null */
```

Comment:
```suggestion
    /** @return string */
```

Reply:
Are you sure? There is a `RETURN_NULL()` at the end of `serialize`.

Reply:
Ooops, you're right. You can drop that condition though, `buf.s` is never NULL there.

Reply:
Roger.